terraform-test/action.yml

```
@@ -0,0 +1,145 @@
+name: "Run terraform test"
+description: |
+  Terraform test for modules.
+author: "Artjoms Laivins"
+
+inputs:
+  test-file:
+    description: Test file to run.
+    required: true
+
+outputs:
+  exit-code:
+    description: The exit code from the terraform test command.
+    value: ${{ steps.run-tests.outputs.exit-code }}
+  summary:
+    description: Summary of the test run.
+    value: ${{ steps.run-tests.outputs.summary }}
+  report:
+    description: Report file of the test run.
+    value: ${{ steps.run-tests.outputs.report }}
+  json:
+    description: JSON output of the test run.
+    value: ${{ steps.run-tests.outputs.json }}
+
+
+runs:
+  using: "composite"
+  steps:
+    - id: check-prereqs
+      shell: bash
+      run: |
+        # Make sure terraform is available
+
+        set -o allexport; source "${{ github.action_path }}/helpers.sh"; set +o allexport;
+
+        start-group "check prerequisite: terraform binary"
+        if ! command -v terraform version &>/dev/null; then
+          log-error "terraform is not available on path, please install using the 'setup-terraform' action!"
+          exit 1
+        else
+          log-info "using $(terraform version)"
+        fi
+        end-group
+
+    - id: azure-login
+      uses: azure/login@v2
+      with:
+         tenant-id: ${{ env.ARM_TENANT_ID }}
+         subscription-id: ${{ env.ARM_SUBSCRIPTION_ID }}
+         client-id: ${{ env.ARM_CLIENT_ID }}   
```

indentation, also add doc in the action description that these mus be set as envs.

```suggestion
        tenant-id: ${{ env.ARM_TENANT_ID }}
        subscription-id: ${{ env.ARM_SUBSCRIPTION_ID }}
        client-id: ${{ env.ARM_CLIENT_ID }}   
```